saleor/tests/fixtures.py

```diff
@@ -3267,6 +3267,26 @@ def product_with_images(product_type, category, media_root, channel_USD):
     return product
 
 
+@pytest.fixture
+def product_with_zero_discount(product_list, product_type, category, channel_USD):
+    product = Product.objects.create(
+        name="Test product 4",
+        slug="test-product-4",
+        product_type=product_type,
+        category=category,
+    )
+    ProductChannelListing.objects.create(
+        product=product,
+        channel=channel_USD,
+        is_published=True,
+        discounted_price_amount=0,
+        currency=channel_USD.currency_code,
+        visible_in_listings=True,
+        available_for_purchase_at=(datetime.datetime(1999, 1, 1, tzinfo=pytz.UTC)),
+    )
+    return product_list
```

Comment:
Why does this fixture take `product_list` and return `product_list` instead of the newly created product?

Reply:
I just thought of reusing the product_list and adding one more product to it to test the bug fix

Reply:
The resulting fixture does not make sense and is not reusable. Instead, I suggest you write a separate test that explicitly discounts the product within the test body so it's clear what the test is checking.

Reply:
sure will do that

Reply:
@patrys I have written a separate test as you suggested, please review it, thanks.

Reply:
@patrys did you get a chance to review the test?

Reply:
I am not the right person to review it, requested a review from the core team.

Reply:
ok thanks